src/main/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImpl.java

```diff
@@ -0,0 +1,371 @@
+        }
+
+        return null;
+    }
+
+    /**
+     * Create DocumentGeneratorConnectionImpl instance containing the document render's request
+     * settings. e.g. Method, Service URL, Body, etc.
+     *
+     * @param accountsType - Account type information: account type, ixbrl's template name, account
+     * @param requestBody - the http request request (json format).
+     * @return {@link DocumentGeneratorConnection} with the document render service settings.
+     */
+    private DocumentGeneratorConnection getDocumentGeneratorConnection(
+        AccountsType accountsType,
+        String requestBody) {
+        DocumentGeneratorConnection connection = new DocumentGeneratorConnection();
+
+        connection.setRequestMethod("POST");
+        connection.setServiceURL(getServiceURL());
+        connection.setRequestBody(requestBody);
+        connection.setAuthorizationProperty(getAPIAuthorization());
+        connection.setAssetId(accountsType.getAssetId());
+        connection.setContentType(MediaType.TEXT_HTML_VALUE);
+        connection.setAcceptType(MediaType.TEXT_HTML_VALUE);
+        connection.setLocation(getIXBRLLocation(accountsType));
+        connection.setTemplateName(accountsType.getTemplateName());
+        connection.setSetDoOutPut(true);
+
+        return connection;
+    }
+
+    /**
+     * Get the location the document render service needs to stored the ixbrl document. e.g.
+     * "s3://dev-pdf-bucket/chs-dev/accounts/small_full_accounts"
+     *
+     * @return
+     */
+    private String getIXBRLLocation(AccountsType accountsType) {
+        return
+            String.format(
+                IXBRL_LOCATION,
+                getMandatoryEnvVariable(DOCUMENT_BUCKET_NAME_ENV_VAR),
+                accountsType.getAssetId(),
+                accountsType.getResourceKey());
+    }
+
+    /**
+     * Get the API key needed to call the service.
+     */
+    private String getAPIAuthorization() {
+        return getMandatoryEnvVariable(API_KEY_ENV_VAR);
+    }
+
+    /**
+     * Build the document render service URL.
+     *
+     * @return the document render service end point e.g. "http://chs-dev:4082/document-render/store"
+     */
+    private String getServiceURL() {
+        return getMandatoryEnvVariable(DOCUMENT_RENDER_SERVICE_HOST_ENV_VAR) +
+            DOCUMENT_RENDER_SERVICE_END_POINT;
+    }
+
+    /**
+     * Get environment variable value for the passed variable.
+     *
+     * @return environment variable value.
+     */
+    private String getMandatoryEnvVariable(String envVariable) {
+        return environmentReader.getMandatoryString(envVariable);
+    }
+
+    /**
+     * Generates the json for small full account.
+     *
+     * @param account - small full's account information.
+     * @return The account marshaled to JSON and converted to String.
+     * @throws JsonProcessingException
+     */
+    private String generateCompanyAccountJSON(Account account)
```

Comment:
This should be called generateSmallFullAccountJSON to match the specific small-full functionality.

Reply:
adf92c4